airflow/models/taskinstance.py

```diff
@@ -1583,14 +2334,22 @@ def _run_raw_task(
 
             # Recording SKIPPED or SUCCESS
             self.clear_next_method_args()
-            self.end_date = timezone.utcnow()
-            self._log_state()
-            self.set_duration()
+            _log_state(task_instance=self)
+            TaskInstance.set_end_date(
+                dag_id=self.dag_id,
+                run_id=self.run_id,
+                task_id=self.task_id,
+                map_index=self.map_index,
+                end_date=timezone.utcnow(),
+                session=session,
+            )
+
+            self.refresh_from_db(session=session)
```

Comment:
I think this is the problem
In line 2276 we set
state= SUCCESS
but we don't commit it . And refresh_from_db simply overrides it to RUNNING.

Do we need to call TaskInstance.set_end_date there? the previous implementation simply set end_date and duration, while you commit the changes (and reload) from DB.

Maybe we should revert TaskInstance.set_end_date to previous implementation (this PR is big enought).
I have on my the task to migrate this (_run_raw_task) so I will think about the possible solutions.

Reply:
I am good with that option :D. Let me revert that then